src/components/elements/generatedManuscriptText.tsx

```diff
@@ -0,0 +1,208 @@
+        let lines = null
+        let folio_start = null
+
+        if (total_folios != null) {
+            folios = <> <b>{total_folios}</b> folios</>
+        }
+
+        if (total_pages != null) {
+            pages = <>, <b>{total_pages}</b> pages</>
+        }
+
+        if (total_scans != null) {
+            scans = <>, and <b>{total_scans}</b> scans</>
+        }
+
+        if (columns_per_page != null) {
+            columns = <><b>{columns_per_page}</b> columns per page</>
+        }
+
+        if (line_range_per_column != null) {
+            lines = <> and approximately <b>{line_range_per_column}</b> lines per column</>
+        }
+
+        if (folio_start_of_the_tm_part != null) {
+            folio_start = <> The Marian miracle stories begin on folio <b>{folio_start_of_the_tm_part}</b> of the whole manuscript.</>
+        }
+
+        return <>This manuscript has a total of {folios}{pages}{scans}. It has {columns} {lines}. {folio_start}</>
+    }
+
+    // duplicate/missing_scans/rebound_in_disorder= COLUMN DOESNT EXIST IN SHEET --> USE UNDERSCORES AND NOT "/"
+    const DeterminePageProblemSentence = (disordered_rebound: string, duplicate_or_missing: string) => {
+        if (disordered_rebound != null && duplicate_or_missing != null) {
+            return <> Unfortunately, this manuscript was bound in disorder and digitized improperly so that some folios were skipped or duplicated.</>
+        }
+        else if (disordered_rebound != null) {
+            return <> Unfortunately, this manuscript was bound in disorder.</>
+        }
+        else if (duplicate_or_missing != null) {
+            return <> Unfortunately, this manuscript was digitized improperly and some folios were skipped or duplicated.</>
+        }
+        return
+    }
+
+    // SEE TICKET ABOUT HYPERLINK TO STAFF PAGE
+    const DetermineCatalogerSentence = (pemm_cataloger_name: string, print_cataloger_name: string, pemm_volunteer_name: string) => {
+        let s1 = null
+        let s2 = null
+        if (pemm_cataloger_name != null && print_cataloger_name != null) {
+            s1 = <> The stories in this manuscript were cataloged by <b>{pemm_cataloger_name}</b> and <b>{print_cataloger_name}</b>.</>
+        }
+        else {
+            if (pemm_cataloger_name != null) {
+                s1 = <> The stories in this manuscript were cataloged by <b>{pemm_cataloger_name}</b></>
+            }
+            else if (print_cataloger_name != null) {
+                s1 = <> The stories in this manuscript were cataloged by <b>{print_cataloger_name}</b></>
+            }
+        }
+
+        if (pemm_volunteer_name != null) {
+            s2 = <> <b>{pemm_volunteer_name}</b> provided assistance by typing incipits.</>
+        }
+
+        return <> {s1} {s2}</>
+    }
+
+    // parts n, m, p instructions are missing from issue post
+    // WHERE TO GET UPLOAD DATE + TOTAL NUM OF MANUSCRIPTS
+    return (
+        <>
+            <div>
+                <h2>
```

Comment:
I think you should hide the sentences if some or all of the data isn't present. so like for the first sentence you can do this: 
```
{
  manuscript.language && manuscript.date_range_start && manuscript.date_range_end &&
  <h2>
      This <u>{manuscript.language}</u> language manuscript was created between {manuscript.date_range_start} and {manuscript.date_range_end}. 
      {DetermineManuscriptDateBasis(manuscript.date_note)}
  </h2>
}
```

Reply:
This will hide the first sentence and the output of `DetermineManuscriptDateBasis` if any of these are undefined: `language`, `date_range_start`,  `date_range_end`. The first sentence doesn't make sense if any of the data is undefined and the output of `DetermineManuscriptDateBasis` doesn't make sense if the first sentence doesn't exist.

Reply:
You should do the same to the other sentences, with the idea that people should have a readable, grammatically correct paragraph regardless of what data is available. Worst case scenario is that it's completely empty because so much data is missing that we've hidden all the sentences, but having no sentences is better than a meaningless/confusing sentence.